input.go

```diff
@@ -83,6 +92,37 @@ func (i *Input) ReadFrom(r io.Reader) (int64, error) {
 	i.UnlockingScript = bscript.NewFromBytes(script)
 	i.SequenceNumber = binary.LittleEndian.Uint32(sequence)
 
+	if extended {
```

Comment:
It would be nice if instead of hardcoding support for additional tx formats / specs we could extract this, perhaps based on tx options that can then have additional format funcs registered? Is that overkill - just worried about catering for future tx spec additions.

Reply:
We are looking for the ReadFrom() method to be able to decipher normal and extended transaction bytes.  As we do not know beforehand what format it is in, the logic is inline within this function.